joblib/test/test_parallel.py

```diff
@@ -1741,3 +1741,25 @@ def test_threadpool_limitation_in_child_context_error(backend):
 
     with raises(AssertionError, match=r"does not acc.*inner_max_num_threads"):
         parallel_backend(backend, inner_max_num_threads=1)
+
+
+@with_multiprocessing
+@parametrize('n_jobs', [2, 4, -1])
+def test_loky_reuse_workers(n_jobs):
+    # Non-regression test for issue #967 where the workers are not reused when
+    # calling multiple Parallel loops.
+
+    def parallel_call(n_jobs):
+        x = range(10)
+        Parallel(n_jobs=n_jobs)(delayed(sum)(x) for i in range(10))
+
+    # Run a parallel loop and get the workers used for computations
+    parallel_call(n_jobs)
+    first_executor = get_reusable_executor(reuse=True)
+
+    # Ensure that the workers are reused for the next calls, as the executor is
+    # not restarted.
+    for _ in range(10):
+        parallel_call(n_jobs)
+        executor = get_reusable_executor(reuse=True)
+        assert executor == first_executor
```

Comment:
Maybe you could extend this test and make the parallel call return the set of pids of worker processes.

Reply:
I am not sure it is a good idea as if we are unlucky, all jobs might be run on the same worker so it means that you need to introduce timers.

Checking that `get_reusable_executor` stays the same is enough I think but I could also check that `executor._processes.keys()` stays the same if you think it would improve the test.

Reply:
Alright. Indeed I tried and I got random results. Maybe adding a small sleep would work but that's fine as it is.